### ISSUE TITLE
packet splicing

### ISSUE THREAD

**Issue by CharmingZhou** [CopernicaMarketingSoftware/AMQP-CPP#138]
auto messageCb = [](const AMQP::Message& message, uint64_t deliveryTag, bool redelivered)
     {                                                      
     ┊   LOG_INFO("message received:%s",message.body());    
     }┊  

When  consumer receive many huge json  message ,the message' packet will splice.

**Comment by EmielBruijntjes** [comment]
Maybe there was a null byte ('\0') inside your message? The %s prints everything up to this null byte. Check the return value of message.bodySize() too.

If it still fails, please report the smallest possible program that demonstrates this problem.

**Comment by CharmingZhou** [comment]
Thank you very much for replying me out of your busy schedule.  I tested so much times and read the closed issue, I realized I should't use message.body() directly.   Here's  right way：
std::string data (message.body(), message.bodySize());
LOGI("message received:"<<data.c_str());

Now I found the received message is right. I  think it's better to use mesage.body() to show right data，otherwise many people will make same mistake.
